### ISSUE TITLE
Add accessibility note to radio-button

### ISSUE THREAD

**Issue by outerpress** [department-of-veterans-affairs/vets-design-system-documentation#3809]
### Duplicate check

- [x] I've searched for any related issues and avoided creating a duplicate issue.

### This update is for:

Component

### What is the name?

Radio button

### What is the nature of this update?

- [ ] How to build this component/pattern
- [ ] When to use this component/pattern
- [ ] When to use something else
- [ ] Usability guidance
- [x] Accessibility considerations
- [ ] Content considerations
- [ ] Implementation
- [ ] Package information
- [ ] Addition to Forms Library documentation
- [ ] Update to existing Forms Library documentation

### What problem does this solve?

During a 508 audit we received feedback:

> The radio button group for "What type of care do you need?" should have the caption "What type of care do you need?". This needs to be read the first time focus is placed on a radio button in the group. So the first radio button that gets focus should read "What type of care do you need? Radio button Amputation care not checked" or something similar. When you arrow to the next radio button in the group it only needs to read the radio button label "Eye Care Radio button not checked". 
 
We updated our radio buttons, but during testing (https://github.com/department-of-veterans-affairs/va.gov-team/issues/99754) we found that Safari reads the "caption" (the fieldset legend) at the end of every radio option when arrowing through them. See recordings in the next section.

@rileyorr found that this is true when this component is used outside of VA.gov. 

So the change requested is to document this in the component's accessibility section:

1. The correct behavior, the legend should only read at the end of the radio option when first tabbing into the group
2. Safari doesn't support this.

Also fyi, this was specifically testing with VO on Mac. I didn't have other SRs to test with.

### Additional Context

Safari reads the legend at the end of every radio option:

https://github.com/user-attachments/assets/f2e4c447-6a1f-481b-a794-e2ff67020afb

Chrome and FireFox read the legend only when first tabbing into the radio group. This was the behavior recommended/expected by the 508 office:

https://github.com/user-attachments/assets/f21070c6-61ed-49d2-9910-344921247ecb

https://github.com/user-attachments/assets/bcd233a8-adb3-4e5f-9195-2148ab98db82

**Comment by rsmithadhoc** [comment]
PR opened for this: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/3931
